.github/workflows/release.yml

```diff
@@ -29,25 +32,146 @@ jobs:
+          squashfs: dist/ota-client-x86_64-v${{ needs.build_python_packages.outputs.OTACLIENT_VERSION }}.squashfs
+          output_dir: dist
+
+      - name: upload artifacts
+        uses: actions/upload-artifact@v4
+        with:
+          name: artifacts-squashfs_x86_64
+          path: |
+            dist/*.squashfs
+            dist/*.patch
+
+  build_squashfs_image_arm64:
+    runs-on: ubuntu-latest
+    needs: build_python_packages
+    steps:
+      - name: checkout
+        uses: actions/checkout@v4
+
+      - name: build squashfs image for arm64
+        uses: ./.github/actions/build_squashfs_image
+        with:
+          platform: linux/arm64
+          platform_suffix: arm64
+          base_image: arm64v8/ubuntu:22.04
+          whl: ${{ needs.build_python_packages.outputs.OTACLIENT_WHL }}
+          version: ${{ needs.build_python_packages.outputs.OTACLIENT_VERSION }}
+          output_squashfs: dist/ota-client-arm64-v${{ needs.build_python_packages.outputs.OTACLIENT_VERSION }}.squashfs
+
+      - name: build paches
+        uses: ./.github/actions/build_patches
+        with:
+          platform_suffix: arm64
+          version: ${{ needs.build_python_packages.outputs.OTACLIENT_VERSION }}
+          squashfs: dist/ota-client-arm64-v${{ needs.build_python_packages.outputs.OTACLIENT_VERSION }}.squashfs
+          output_dir: dist
+
+      - name: upload artifacts
+        uses: actions/upload-artifact@v4
+        with:
+          name: artifacts-squashfs_arm64
+          path: |
+            dist/*.squashfs
+            dist/*.patch
+
+  post_build:
+    runs-on: ubuntu-latest
+    needs: [build_python_packages, build_squashfs_image_x86_64, build_squashfs_image_arm64]
+    steps:
+      - name: checkout
+        uses: actions/checkout@v4
+
+      - name: download python packages artifact
+        uses: actions/download-artifact@v4
+        with:
+          name: artifacts-python-packages
+          path: dist
+
+      - name: download squashfs x86_64 artifact
+        uses: actions/download-artifact@v4
+        with:
+          name: artifacts-squashfs_x86_64
+          path: dist
+
+      - name: download squashfs arm64 artifact
+        uses: actions/download-artifact@v4
+        with:
+          name: artifacts-squashfs_arm64
+          path: dist
+
+      - name: generate manifest
+        uses: ./.github/actions/generate_manifest
+        with:
+          dir: dist
+          output_manifest: dist/manifest.json
+
       - name: calculate checksum
         run: |
-          for WHL in dist/*.whl; \
-            do \
-              sha256sum ${WHL} | sed -E "s@(\w+)\s+.*@sha256:\1@" > \
-                ${WHL}.checksum; \
-            done
+          for FILE in dist/*.{whl,squashfs,patch,json}; do
+            if [ -s ${FILE} ]; then
+              sha256sum ${FILE} | sed -E "s@(\w+)\s+.*@sha256:\1@" > ${FILE}.checksum
+            fi
+          done
```

Comment:
Since we have already calculated the checksum when building manifest.json, what about just let generate_manifest action to also generate the checksum files, so that we don't need dedicated step for checksum calculation.

Reply:
`manifest.json` is created only for `.squashfs` and `.patch`, not include `.whl`.
Thus my suggestion is to separate the manifest generation and checksum steps like current implementation, and calculate checksum for all of the artifacts.

Reply:
Understood, thanks!